src/main/java/com/cburch/logisim/fpga/hdlgenerator/WithSelectHDLGenerator.java

```diff
@@ -0,0 +1,117 @@
+ *
+ * https://github.com/logisim-evolution/
+ *
+ * This is free software released under GNU GPLv3 license
+ */
+package com.cburch.logisim.fpga.hdlgenerator;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.cburch.logisim.util.LineBuffer;
+
+public class WithSelectHDLGenerator {
+  
+  private final Map<Long, Long> myCases;
+  private final String regName;
+  private final String sourceSignal;
+  private final int nrOfSourceBits;
+  private final String destinationSignal;
+  private final int nrOfDestinationBits;
+  private Long defaultValue = 0L;
+  
+  public WithSelectHDLGenerator(String componentName, String sourceSignal, int nrOfSourceBits, 
+      String destinationSignal, int nrOfDestinationBits) {
+    myCases = new HashMap<>();
+    regName = LineBuffer.format("s_{{1}}_reg", componentName);
+    this.sourceSignal = sourceSignal;
+    this.nrOfSourceBits = nrOfSourceBits;
+    this.destinationSignal = destinationSignal;
+    this.nrOfDestinationBits = nrOfDestinationBits;
+  }
+  
+  private Long binairyStringToInt(String binairyValue) {
+    var result = 0L;
+    for (var charIndex = 0; charIndex < binairyValue.length(); charIndex++) {
+      final var character = binairyValue.charAt(charIndex) - '0';
+      if ((character < 0) || (character > 1)) throw new NumberFormatException("Invalid binairy value in WithSelectHDLGenerator");
+      result *= 2;
+      result += character;
+    }
+    return result;
+  }
+  
+  public WithSelectHDLGenerator add(Long selectValue, Long assignValue) {
+    myCases.put(selectValue, assignValue);
+    return this;
+  }
+  
+  public WithSelectHDLGenerator add(Long selectValue, String binairyAssignValue) {
+    myCases.put(selectValue, binairyStringToInt(binairyAssignValue));
+    return this;
+  }
+  
+  public WithSelectHDLGenerator add(String binairySelectValue, String binairyAssignValue) {
+    myCases.put(binairyStringToInt(binairySelectValue), binairyStringToInt(binairyAssignValue));
+    return this;
+  }
+  
+  public WithSelectHDLGenerator add(String binairySelectValue, Long assignValue) {
+    myCases.put(binairyStringToInt(binairySelectValue), assignValue);
+    return this;
+  }
+  
+  public WithSelectHDLGenerator setDefault(Long assignValue) {
+    defaultValue = assignValue;
+    return this;
+  }
+  
+  public WithSelectHDLGenerator setDefault(String binairyAssignValue) {
+    defaultValue = binairyStringToInt(binairyAssignValue);
+    return this;
+  }
+  
+  public ArrayList<String> getHdlCode() {
+    final var contents = (new LineBuffer()).withHdlPairs()
+        .pair("sourceName", sourceSignal)
+        .pair("destName", destinationSignal)
+        .pair("regName", regName)
+        .pair("regBits", nrOfDestinationBits - 1);
+    contents.add("");
+    if (HDL.isVHDL()) {
+      contents.add("WITH ({{sourceName}}) SELECT {{destName}} <=");
+    } else {
+      contents.add("""
+        reg[{{regBits}}:0] {{regName}};
```

Comment:
This line indentation looks off - see indentation of closing `"""`.

Reply:
fixed

Reply:
Can you double check it's fixed, because I cannot see that:

![ss](https://user-images.githubusercontent.com/8041294/132956586-75ab1dcf-291d-4533-9750-cfe7567f92d2.png)